Bot.Core/TownMaintenance.cs

```diff
@@ -44,13 +46,19 @@ public TownMaintenance(IServiceProvider sp)
             m_shutdownPrevention.RegisterShutdownPreventer(m_shutdownTcs.Task);
 
             var callbackFactory = sp.GetService<ICallbackSchedulerFactory>();
+            // Scheduler for when we're running the queue - every few minutes, process some more towns
             m_callbackScheduler = callbackFactory.CreateScheduler<Queue<TownKey>>(ProcessQueue, TimeSpan.FromMinutes(1));
+            // Scheduler for when we're waiting between maintenance periods - only needs to check every day or so
+            m_longWaitScheduler = callbackFactory.CreateScheduler<bool>(RunMaintenanceAsync, TimeSpan.FromDays(1));
```

Comment:
We could probably easily make a secondary type of scheduler that doesn't take any params.